src/shogun/multiclass/KNN.cpp

```diff
@@ -575,55 +315,32 @@ void CKNN::store_model_features()
 	SG_UNREF(d_rhs);
 }
 
-int32_t CKNN::choose_class(float64_t* classes, int32_t* train_lab)
+void CKNN::init_solver(KNN_SOLVER knn_solver)
```

Comment:
mmmm i'm not so sure if i follow.
you can simply manually run valgrind on the knn examples/unit tests and you'll see how it is leaking.